docs/APP-SNAPSHOT.md

```diff
@@ -0,0 +1,1053 @@
+    * [pkg/pillar/volumehandlers](../pkg/pillar/volumehandlers)
+    * [pkg/pillar/diskmetrics/diskmetrics.go](../pkg/pillar/diskmetrics/diskmetrics.go)
+6. **Volumemgr to Zedmanager**: Once the snapshot action is performed, the
+   volumemgr sends a VolumesSnapshotStatus back to the zedmanager. This status
+   includes information about the snapshot, such as the snapshot ID, metadata,
+   time created, app UUID, reference count, the result of the action, and error
+   information. Sometimes to report an error on this step, a dedicated status
+   can be created. It’s done when there is no existing one.
+    * [pkg/pillar/cmd/volumemgr/handlesnapshot.go](../pkg/pillar/cmd/volumemgr/handlesnapshot.go)
+7. **Zedmanager: Reaction** \
+   The zedmanager receives the VolumesSnapshotStatus from the volumemgr and
+   updates the SnapshottingStatus in the AppInstanceStatus. It handles any
+   further actions based on the updated status and reports the updated status
+   back to the zedagent.
+    * [pkg/pillar/cmd/zedmanager/handlevolumemgr.go](../pkg/pillar/cmd/zedmanager/handlevolumemgr.go)
+    * [pkg/pillar/cmd/zedmanager/updatestatus.go](../pkg/pillar/cmd/zedmanager/updatestatus.go)
+8. **Zedagent to Controller**: The zedagent receives the updated
+   AppInstanceStatus from the zedmanager and reports the updated status,
+   including the SnapshottingStatus, back to the controller.
+    * [pkg/pillar/cmd/zedagent/handlemetrics.go](../pkg/pillar/cmd/zedagent/handlemetrics.go)
+
+This data flow ensures that the requests from the controller are processed and
+handled by the appropriate components and that the controller is informed of the
+updated status of the app instance and any associated snapshots. The stored
+information in the VolumesSnapshotStatus and SnapshottingStatus can be used for
+future operations related to app instances and snapshots.
+
+## Snapshot Data Structures
+
+The snapshot creation process relies on several data structures to manage and
+track the status of snapshots for each application instance. These data
+structures are part of the `AppInstanceStatus` and are used
+to handle snapshot requests, creation, deletion, and rollback.
+
+### SnapshottingStatus
+
+This is the main data structure that manages and tracks all snapshot-related
+activities for an app instance. It is included in the AppInstanceStatus
+structure. It contains fields that track the state of snapshots for the app
+instance, such as RequestedSnapshots, AvailableSnapshots, and
+SnapshotsToBeDeleted. It also includes fields that track the state of
+snapshot-related actions, such as SnapshotOnUpgrade, HasRollbackRequest,
+ActiveSnapshot, and RollbackInProgress.
+
+It is updated by the zedmanager based on snapshot requests from the controller,
+snapshot status from the volumemgr, and snapshot-related actions performed by
+the zedmanager.
+
+Defined
+in [pkg/pillar/types/zedmanagertypes.go](../pkg/pillar/types/zedmanagertypes.go).
+
+### SnapshotInstanceStatus
```

Comment:
It might make sense to add a note in this section that this struct is serialized to have it persist across device restarts, hence care needs to be taken when updating the struct definition in future EVE releases.

Reply:
I have added a node here and also to the `VolumesSnapshotStatus`.